docs/usage/user.asciidoc

```diff
@@ -0,0 +1,396 @@
+
+[source,json]
+-----------
+{
+  "user": {
+    "name": "alice",
+    "id": "1001",
+    "effective": {
+      "name": "root",
+      "id": "1"
+    },
+    "target": {
+      "name": "bob",
+      "id": "1002"
+    },
+    "changes": {
+      "name": "bob.barker"
+    }
+  },
+  "related": { "user": ["alice", "bob", "bob.barker", "root"] }
+}
+-----------
+
+[discrete]
+[[ecs-user-usage-reuse-subtleties]]
+====== Subtleties around field reuse
+
+Most cases of field reuse in ECS are reusing a field set inside another field set.
+Two examples of this are:
+
+* reusing `group` in `user`, resulting in the `user.group.*` fields, or
+* reusing `user` in `destination`, resulting in the `destination.user.*` fields,
+  which also include `destination.user.group.*`.
+
+The `user` fields can also be reused within `user` as different names,
+representing the role of each relevant user. Examples are the `user.target.*` or `user.effective.*` fields.
+
+It's important to note, that contrary to the `group` fields,
+the user fields reused within `user` are **not** carried around when reusing `user`
+in other places. To continue with the `destination` example, while `group` fields
+are carried to `destination.user.group.*`, there are no `destination.user.effective.*`,
+`destination.user.target.*` nor `destination.user.changes.*` fields.
+
+[discrete]
+[[ecs-user-usage-pivoting]]
+===== Pivoting via related.user
+
+In all events in this page, we've populated the `related.user` fields.
+
+Any event that has users in it should always populate the array field `related.user`
+with all usernames seen in the event; including event names that appear in custom fields.
+Note that this field is not a nesting of all user fields,
+it's a flat array meant to contain user identifiers.
+
+Taking the example from `user.changes` again, we can see that no matter the role
+of the each user (before/after privilege escalation, affected user, username after rename), they are all present in `related.user`:
+
+[source,json]
+-----------
+{
+  "user": {
+    "name": "alice",
+    "id": "1001",
+    "effective": {
+      "name": "root",
+      "id": "1"
+    },
+    "target": {
+      "name": "bob",
+      "id": "1002"
+    },
+    "changes": {
+      "name": "bob.barker"
+    }
+  },
+  "related": { "user": ["alice", "root", "bob", "bob.barker"] }
+}
+-----------
+
+Like the other fields in the <<ecs-related,related>> field set, `related.user` is meant to facilitate
+pivoting. For example, if you have a suspicion about user "bob.barker", searching
+for this name in `related.user` will give you all events related to this user, whether
+it's the creation / rename of the user, or events where this user was active in a system.
+
+[discrete]
+[[ecs-user-usage-mappings]]
+===== Mapping Examples
+
+For examples of mapping events from various sources, you can look at
+https://github.com/elastic/ecs/blob/master/rfcs/text/0007-multiple-users.md#source-data[RFC 0007 in section Source Data].
```

Comment:
It's neither in-scope nor a blocker for this PR, but I'd like to add project-wide ECS attributes to our docs for commonly (or potentially common) reused strings such as the ECS repo GitHub URL.

We will appreciate it when/if we need to update the path to only have the single place 😂

Reply:
Yeah very good idea. It will be useful when we tackle inclusive naming.